lib/reline.rb

```diff
@@ -400,9 +390,8 @@ def readline(prompt = '', add_hist = false)
         end
         case result
         when :matched
-          expanded = key_stroke.expand(buffer).map{ |expanded_c|
-            Reline::Key.new(expanded_c, expanded_c, false)
-          }
+          expanded, rest_bytes = key_stroke.expand(buffer)
+          rest_bytes.reverse_each { |c| io_gate.ungetc(c) }
```

Comment:
This operation appears 3 times. Maybe we can put it inside a private method and use a name to explain what it does?

Reply:
Actually, I leaved it as a duplicated code to make the scope of this pull request small.
In the next pull request #712, many`block.(Key.new)` and other two `key_stroke.expand` `rest_bytes.reverse_each` will disappear.
https://github.com/ruby/reline/pull/712/commits/2fe13a12ae118a25d010bc36e91ec81beb605dcf